project4/name_server/tasks/bootstrap_task.h

```diff
@@ -0,0 +1,40 @@
+/**
+ * @file Bootstrap Task
+ */
+#ifndef PROJECT4_BOOTSTRAP_TASK_H
+#define PROJECT4_BOOTSTRAP_TASK_H
+
+#include <iostream>
+#include "thread_pool/task.h"
+#include "../bootstrap.h"
+#include "../commands.h"
+#include "queue/queue.h"
+
+namespace nameserver::tasks {
+
+/**
+ * @class The task that controls the contained bootstrap
+ */
+class BootstrapTask : public thread_pool::Task {
+ public:
+
+  /**
+   * @brief Initializes the bootstrap task based on config file.
+   * @param config_file the config file of this bootstrap
+   */
+  explicit BootstrapTask(const std::filesystem::path config_file);
+
+  Status SetUp() override;
+
+  Status RunAtomic() override;
+
+ private:
+  /// Bootstrap
+  nameserver::Bootstrap bootstrap_;
+
+  /// Command queue
+  queue::Queue<std::pair<nameserver::BootstrapCommand, std::string>> cmd_queue_;
```

Comment:
This will almost certainly end up having to be a `shared_ptr`, because for a queue to be useful, it has to be shared between multiple places.